### PR TITLE
Controller layout alternative

### PR DESCRIPTION
**Pull request by jnallard**
This changes auto detects the controller that's currently in the driver port and will handle if the controller type switches while the simulator is active.

image

### CODE REVIEW

File: src/main/java/frc/robot/gamepads/XboxControllerMapping.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            case B:
+                return joystick.getRawButton(2);
+            case X:
+                return joystick.getRawButton(3);
+            case Y:
+                return joystick.getRawButton(4);
+            case LeftBumper:
+                return joystick.getRawButton(5);
+            case LeftTrigger:
+                return joystick.getRawAxis(4) > 0.5;
```
Comment (jnallard) [inline src/main/java/frc/robot/gamepads/XboxControllerMapping.java:45]: Had to make the code a bit more complicated than originally wanted. 
The Xbox controller treats the triggers as axes, but the logitech treats them as buttons.

File: src/main/java/frc/robot/gamepads/Gamepad.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    private Button createButton(ButtonType buttonType) {
+        return new Button() {
+            public boolean get() {
+                return getControllerMapping().getButtonPressed(m_joystick, buttonType);
+            }
+        };
+    }
+
+    private void addShuffleboeardTab() {
+        ShuffleboardTab tab = Shuffleboard.getTab("Gamepad - " + m_controllerTitle);
```
Comment (jnallard) [inline src/main/java/frc/robot/gamepads/Gamepad.java:132]: Added code to show the buttons on the shuffle board to easily debug problems. 
Posted a screenshot in the PR description.

File: src/main/java/frc/robot/commands/TankDrive.java [hunk 3]
```diff
@@ -23,15 +22,19 @@ public TankDrive(Joystick Driver, RomiDrivetrain Drivetrain) {
...
   @Override
-  public void initialize() {}
+  public void initialize() {
+  }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
-  public void execute() {m_Drivetrain.TankDrive(m_Driver.getRawAxis(5), m_Driver.getRawAxis(1));}
+  public void execute() {
+    m_Drivetrain.TankDrive(m_Driver.getLeftY(), m_Driver.getRightY());
```
Comment (aimiebell) [inline src/main/java/frc/robot/commands/TankDrive.java:31]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Review by BenColby6 — APPROVED** [review]
Reviewed together.

**Pull request merged**